examples/webbilling-demo/src/tests/helpers/test-helpers.ts

```diff
@@ -179,8 +181,17 @@ export async function enterCreditCardDetails(
 
   page.locator("button[data-testid='PayButton']").waitFor();
   const checkoutTitle = page.getByText("Secure Checkout");
-
   await expect(checkoutTitle).toBeVisible();
+
+  const allowingCardUsageText = page.getByText(
+    /By providing your card information you allow/,
+  );
+  if (isSubscription) {
+    await expect(allowingCardUsageText).toBeVisible();
+  } else {
+    await expect(allowingCardUsageText).not.toBeVisible();
+  }
```

Comment:
Checking this in every single E2E test seems excessive to me. Since you’ve added a story for each product type, I think that should be sufficient.

If we still want to verify this in E2E tests, I’d suggest introducing helper methods like `confirmTermsVisible(page: Page)` and `confirmTermsNotVisible(page: Page)`, and only checking this in the following tests:
- Purchase a subscription product
- Purchase a consumable product
- Purchase a non-consumable product

Reply:
Removing the tests